### ISSUE TITLE
Only H2 driver shows up with Java 9

### ISSUE THREAD

**Issue by wamdam** [metabase/metabase#2881]
-  My computer's OS is Ubuntu 16.04.
-  I'm running Metabase locally from the JAR with openjdk 9
-  My Metabase version is v0.18.1 (9f3beac release-0.18.1).

```
java -version
openjdk version "9-internal"
OpenJDK Runtime Environment (build 9-internal+0-2016-04-14-195246.buildd.src)
OpenJDK 64-Bit Server VM (build 9-internal+0-2016-04-14-195246.buildd.src, mixed mode)
```

```
 java -jar metabase.jar 
07-05 08:37:17 INFO metabase.core :: Starting Metabase in STANDALONE mode
07-05 08:37:17 INFO metabase.core :: Launching Embedded Jetty Webserver with config:
 {:port 3000}

07-05 08:37:17 INFO metabase.core :: Starting Metabase version v0.18.1 (9f3beac release-0.18.1) ...
07-05 08:37:17 INFO metabase.core :: System timezone is 'Europe/Berlin' ...
07-05 08:37:17 INFO metabase.db :: Verifying Database Connection ...
07-05 08:37:17 DEBUG metabase.driver :: Registered driver :h2 ??
07-05 08:37:20 INFO metabase.db :: Verify Database Connection ... ✅
07-05 08:37:45 INFO metabase.db :: Database Migrations Current ... ✅
07-05 08:37:46 INFO db.migrations :: Running all necessary data migrations, this may take a minute.
07-05 08:37:47 INFO db.migrations :: Finished running data migrations.
07-05 08:37:47 INFO metabase.core :: Metabase Initialization COMPLETE
```

However, when I explicitly define MB_DB_\* for e.g. postgresql, then the postgresql driver is available.

THIS IS ONLY FOR OPENJDK 1.9.

For openjdk 1.8 as in

```
 java -version
openjdk version "1.8.0_91"
OpenJDK Runtime Environment (build 1.8.0_91-8u91-b14-0ubuntu4~16.04.1-b14)
OpenJDK 64-Bit Server VM (build 25.91-b14, mixed mode)
```

I get the correct behavour:

```
 java -jar metabase.jar 
07-05 09:26:43 INFO metabase.core :: Starting Metabase in STANDALONE mode
07-05 09:26:43 INFO metabase.core :: Launching Embedded Jetty Webserver with config:
 {:port 3000}

07-05 09:26:43 INFO metabase.core :: Starting Metabase version v0.18.1 (9f3beac release-0.18.1) ...
07-05 09:26:43 INFO metabase.core :: System timezone is 'Europe/Berlin' ...
07-05 09:26:44 DEBUG metabase.driver :: Registered driver :bigquery ??
07-05 09:26:44 DEBUG metabase.driver :: Registered driver :crate ??
07-05 09:26:44 DEBUG metabase.driver :: Registered driver :druid ??
07-05 09:26:44 DEBUG metabase.driver :: Registered driver :h2 ??
07-05 09:26:44 DEBUG metabase.driver :: Registered driver :mongo ??
07-05 09:26:44 DEBUG metabase.driver :: Registered driver :mysql ??
07-05 09:26:44 DEBUG metabase.driver :: Registered driver :postgres ??
07-05 09:26:44 DEBUG metabase.driver :: Registered driver :redshift ??
07-05 09:26:44 DEBUG metabase.driver :: Registered driver :sqlite ??
07-05 09:26:44 DEBUG metabase.driver :: Registered driver :sqlserver ??
07-05 09:26:45 INFO metabase.db :: Verifying Database Connection ...
07-05 09:26:46 INFO metabase.db :: Verify Database Connection ... ✅
07-05 09:27:07 INFO metabase.db :: Database Migrations Current ... ✅
07-05 09:27:07 INFO db.migrations :: Running all necessary data migrations, this may take a minute.
07-05 09:27:07 INFO db.migrations :: Finished running data migrations.
07-05 09:27:07 INFO metabase.events :: Starting events listener: metabase.events.activity-feed ??
07-05 09:27:07 INFO metabase.events :: Starting events listener: metabase.events.dependencies ??
07-05 09:27:07 INFO metabase.events :: Starting events listener: metabase.events.driver-notifications ??
07-05 09:27:07 INFO metabase.events :: Starting events listener: metabase.events.last-login ??
07-05 09:27:07 INFO metabase.events :: Starting events listener: metabase.events.metabot-lifecycle ??
07-05 09:27:07 INFO metabase.events :: Starting events listener: metabase.events.notifications ??
07-05 09:27:07 INFO metabase.events :: Starting events listener: metabase.events.revision ??
07-05 09:27:07 INFO metabase.events :: Starting events listener: metabase.events.sync-database ??
07-05 09:27:07 INFO metabase.events :: Starting events listener: metabase.events.view-log ??
07-05 09:27:08 INFO metabase.task :: Loading tasks namespace: metabase.task.follow-up-emails ??
07-05 09:27:08 INFO metabase.task :: Loading tasks namespace: metabase.task.send-pulses ??
07-05 09:27:08 INFO metabase.task :: Loading tasks namespace: metabase.task.sync-databases ??
07-05 09:27:08 INFO metabase.task :: Loading tasks namespace: metabase.task.upgrade-checks ??
07-05 09:27:08 INFO metabase.core :: Metabase Initialization COMPLETE
```

It seems like the code for find-and-load-drivers in metabase/driver.clj does not work for openjdk 1.9, as well as probably the code for finding metabase.events.

For now, I'd recommend to explicitly tell which java versions are known to work at http://www.metabase.com/docs/v0.18.1/operations-guide/running-the-metabase-jar-file.html

**Comment by salsakran** [comment]
Given that openjdk 9 isn't going to have a release candidate until January 
http://openjdk.java.net/projects/jdk9/ , I'm marking this as won't fix.

If once an RC is out, this is still broken due to a fundamental shift in Java 9 vs a possible bug in the jdk, we'll definitely fix it then. 

To say we don't recommend running Metabase in production on pre-RC versions of our dependencies would be an understatement =)

**Comment by inkrement** [comment]
I get the same error using the official openjava-9 release (only the h2 driver is available).

**Comment by camsaul** [comment]
Going to reopen this now that Java 9 is officially out and this still isn't working

**Comment by camsaul** [comment]
Fixed by #6211
